scripts/tests/twister_blackbox/test_coverage.py

```diff
@@ -0,0 +1,294 @@
+            [
+                'coverage.log', 'coverage.json',
+                'coverage', os.path.join('coverage','coverage.txt')
+            ],
+            'txt'
+        ),
+        (
+            'gcovr',
+            [
+                'coverage.log', 'coverage.json',
+                'coverage', os.path.join('coverage','coverage.csv')
+            ],
+            'csv'
+        ),
+        (
+            'gcovr',
+            [
+                'coverage.log', 'coverage.json',
+                'coverage', os.path.join('coverage','coverage.coveralls.json')
+            ],
+            'coveralls'
+        ),
+        (
+            'gcovr',
+            [
+                'coverage.log', 'coverage.json',
+                'coverage', os.path.join('coverage','index.html')
+            ],
+            'html'
+        ),
+        (
+            'lcov',
+            [
+                'coverage.log', 'coverage.info',
+                'ztest.info', 'coverage',
+                os.path.join('coverage','index.html')
+            ],
+            'html'
+        ),
+        (
+            'lcov',
+            [
+                'coverage.log', 'coverage.info',
+                'ztest.info'
+            ],
+            'lcov'
+        ),
+    ]
+    TESTDATA_5 = [
+        (
+            os.path.join(TEST_DATA, 'tests', 'dummy', 'agnostic', 'group2'),
+            ['qemu_x86'],
+            'gcovr',
+            'Running gcovr -r'
+        ),
+        (
+            os.path.join(TEST_DATA, 'tests', 'dummy', 'agnostic', 'group2'),
+            ['qemu_x86'],
+            'lcov',
+            'Running lcov --gcov-tool'
+        )
+    ]
+
+    @classmethod
+    def setup_class(cls):
+        apath = os.path.join(ZEPHYR_BASE, 'scripts', 'twister')
+        cls.loader = importlib.machinery.SourceFileLoader('__main__', apath)
+        cls.spec = importlib.util.spec_from_loader(cls.loader.name, cls.loader)
+        cls.twister_module = importlib.util.module_from_spec(cls.spec)
+
+    @pytest.mark.parametrize(
+        'test_path, test_platforms, file_name',
+        TESTDATA_1,
+        ids=[
+            'coverage',
+        ]
+    )
+    def test_coverage(self, capfd, test_path, test_platforms, out_path, file_name):
+        args = ['-i','--outdir', out_path, '-T', test_path, '--coverage', '--coverage-tool', 'gcovr'] + \
+               [val for pair in zip(
+                   ['-p'] * len(test_platforms), test_platforms
+               ) for val in pair]
+
+        with mock.patch.object(sys, 'argv', [sys.argv[0]] + args), \
+                pytest.raises(SystemExit) as sys_exit:
+            self.loader.exec_module(self.twister_module)
+
+        out, err = capfd.readouterr()
+        sys.stdout.write(out)
+        sys.stderr.write(err)
+
+        for f_name in file_name:
+            path = os.path.join(out_path, f_name)
+            assert os.path.exists(path), f'file not found {f_name}'
+
+        assert str(sys_exit.value) == '0'
```

Comment:
Best to check the exit value in the first assert (if that breaks, you know more about the error).